### PR TITLE
Add restore for CC databases / tables

### LINKED ISSUE #8483: Add Restore Table/Database module guidance to CockroachCloud docs
Recovery Information link (table/database):
- Link opens modal with information for push-button RESTORE
- User pushes button to RESTORE
- User checks status on the CC Ops List Page (and is kicked there once job starts)

### PR DESCRIPTION
**Pull request by lnhsingh**
Closes #8483.

### CODE REVIEW

**Review by mwang1026 — CHANGES_REQUESTED** [review]
A few minor change requests

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/9292#-:-MQNuUnf6EI1TzR_fFKT:b2hey9p)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @j-low, @lnhsingh, and @mwang1026)

___
*[cockroachcloud/backups-page.md, line 16 at r1](https://reviewable.io/reviews/cockroachdb/docs/9292#-MQNVaxtDDCFurJEh449:-MQNVaxtDDCFurJEh44A:b-k0am9) ([raw file](https://github.com/cockroachdb/docs/blob/d9792e20a4d1827715f4453913fdd3eeb9089b43/cockroachcloud/backups-page.md#L16)):*
> ```
> Currently, you cannot access the backups that Cockroach Labs runs to restore your data. In the future, you will be able to restore databases and tables from the CockroachCloud Console. If you need to access these backups, please [contact Support](https://support.cockroachlabs.com/hc/en-us).
> ```

Guess we probably don't need this anymore? :D. Or we have the caveat instead that you can only restore to the same cluster that the backup was taken from.
___
*[cockroachcloud/backups-page.md, line 82 at r1](https://reviewable.io/reviews/cockroachdb/docs/9292#-MQNW8744cHcL-MOLtyN:-MQNW8744cHcL-MOLtyO:b-6h4wh7) ([raw file](https://github.com/cockroachdb/docs/blob/d9792e20a4d1827715f4453913fdd3eeb9089b43/cockroachcloud/backups-page.md#L82)):*
> ```Markdown
> 
> {{site.data.alerts.callout_info}}
> At this time, restoring databases and tables in the Console is only available to CockroachCloud clusters running on GCP.
> ```

See note that I cc'ed you in in Slack. Seems we fixed the AWS stuffs. (although we're currently fighting and issue on this while trying to release)
___
*[cockroachcloud/backups-page.md, line 111 at r1](https://reviewable.io/reviews/cockroachdb/docs/9292#-MQNtEkz75eflnnLel0Q:-MQNtEkz75eflnnLel0R:b-qk11n7) ([raw file](https://github.com/cockroachdb/docs/blob/d9792e20a4d1827715f4453913fdd3eeb9089b43/cockroachcloud/backups-page.md#L111)):*
> ```
>     - **Skip missing foreign keys**, which will remove [foreign key](../stable/foreign-key.html) constraints before restoring.
>     - **Skip missing sequences**, which will ignore [sequence](../stable/show-sequences.html) dependencies (i.e., the `DEFAULT` expression that uses the sequence).
>     - **Skip missing views**, which will skip restoring [views](../stable/views.html) that cannot be restored because their dependencies are not being restored at the same time.
> ```

Formatting of this is weird visually
___
*[cockroachcloud/backups-page.md, line 168 at r1](https://reviewable.io/reviews/cockroachdb/docs/9292#-MQNuCJb3EwNZqZO06Gz:-MQNuCJb3EwNZqZO06H-:bfd4cqt) ([raw file](https://github.com/cockroachdb/docs/blob/d9792e20a4d1827715f4453913fdd3eeb9089b43/cockroachcloud/backups-page.md#L168)):*
> ```Markdown
> 
>     {{site.data.alerts.callout_danger}}
>     If you are backing up the data to AWS or GCP, use the `specified` option for the `AUTH` parameter.
> ```

Technically this only matters for Cockroach Cloud. IF you're self hosting then it's fine if you want to use implicit auth. But on RESTORE you'll need to you `specified` credentials, so maybe worth keeping?

**Review by lnhsingh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/9292)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @j-low and @mwang1026)

___
*[cockroachcloud/backups-page.md, line 16 at r1](https://reviewable.io/reviews/cockroachdb/docs/9292#-MQNVaxtDDCFurJEh449:-MQOuxwh5KEclKt5Zpd2:b-ijo2vu) ([raw file](https://github.com/cockroachdb/docs/blob/d9792e20a4d1827715f4453913fdd3eeb9089b43/cockroachcloud/backups-page.md#L16)):*
<details><summary><i>Previously, mwang1026 (Michael Wang) wrote…</i></summary><blockquote>

> ```
> Currently, you cannot access the backups that Cockroach Labs runs to restore your data. In the future, you will be able to restore databases and tables from the CockroachCloud Console. If you need to access these backups, please [contact Support](https://support.cockroachlabs.com/hc/en-us).
> ```

Guess we probably don't need this anymore? :D. Or we have the caveat instead that you can only restore to the same cluster that the backup was taken from.
</blockquote></details>

Woo woo! Changed to the caveat you mentioned.
___
*[cockroachcloud/backups-page.md, line 82 at r1](https://reviewable.io/reviews/cockroachdb/docs/9292#-MQNW8744cHcL-MOLtyN:-MQOv5MzDcnBsALn-yFk:bss4a2b) ([raw file](https://github.com/cockroachdb/docs/blob/d9792e20a4d1827715f4453913fdd3eeb9089b43/cockroachcloud/backups-page.md#L82)):*
<details><summary><i>Previously, mwang1026 (Michael Wang) wrote…</i></summary><blockquote>

See note that I cc'ed you in in Slack. Seems we fixed the AWS stuffs. (although we're currently fighting and issue on this while trying to release)
</blockquote></details>

Removed.
___
*[cockroachcloud/backups-page.md, line 111 at r1](https://reviewable.io/reviews/cockroachdb/docs/9292#-MQNtEkz75eflnnLel0Q:-MQOvqce-plk3niYcJc9:bt2vr1b) ([raw file](https://github.com/cockroachdb/docs/blob/d9792e20a4d1827715f4453913fdd3eeb9089b43/cockroachcloud/backups-page.md#L111)):*
<details><summary><i>Previously, mwang1026 (Michael Wang) wrote…</i></summary><blockquote>

> ```
>     - **Skip missing foreign keys**, which will remove [foreign key](../stable/foreign-key.html) constraints before restoring.
>     - **Skip missing sequences**, which will ignore [sequence](../stable/show-sequences.html) dependencies (i.e., the `DEFAULT` expression that uses the sequence).
>     - **Skip missing views**, which will skip restoring [views](../stable/views.html) that cannot be restored because their dependencies are not being restored at the same time.
> ```

Formatting of this is weird visually
</blockquote></details>

Fixed.
___
*[cockroachcloud/backups-page.md, line 168 at r1](https://reviewable.io/reviews/cockroachdb/docs/9292#-MQNuCJb3EwNZqZO06Gz:-MQOw8gYBergc4wDmw79:b-n4vnzq) ([raw file](https://github.com/cockroachdb/docs/blob/d9792e20a4d1827715f4453913fdd3eeb9089b43/cockroachcloud/backups-page.md#L168)):*
<details><summary><i>Previously, mwang1026 (Michael Wang) wrote…</i></summary><blockquote>

Technically this only matters for Cockroach Cloud. IF you're self hosting then it's fine if you want to use implicit auth. But on RESTORE you'll need to you `specified` credentials, so maybe worth keeping?
</blockquote></details>

Added more info to clarify. PTAL

**Comment by lnhsingh** [comment]
@mwang1026 moving this to docs review but please feel free to take another look

File: cockroachcloud/backups-page.md [hunk 1]
```diff
@@ -69,19 +71,95 @@ For each table in the database, the following details display:
...
+    The **Databases** view displays.
+
+1. Click **Restore** for the database you want to restore.
+
+    The **Restore database** module displays with backup details.
+
+1. In the **Restore to** field, enter the name of the destination database.
+
+    {{site.data.alerts.callout_info}}
+    [Resolve any naming conflicts](#resolve-a-database-naming-conflict) by using `DROP` or `RENAME` on the existing database. If you enter a unique name in the **Restore to** field, a new database will be created.
```
Comment (rmloveland) [inline cockroachcloud/backups-page.md:101]: links to `DROP DATABASE` and `RENAME DATABASE` ?

File: cockroachcloud/backups-page.md [hunk 2]
```diff
@@ -69,19 +71,95 @@ For each table in the database, the following details display:
...
+- [Back up a self-hosted CockroachDB cluster and restore into a CockroachCloud cluster](#back-up-a-self-hosted-cockroachdb-cluster-and-restore-into-a-cockroachcloud-cluster)
+- [Back up and restore data manually](#back-up-and-restore-data-manually)
+
+### Restore a database
+
+To restore a database:
+
+1. Find the cluster backup containing the database you want to restore, and click the number in the corresponding **Databases** column.
+
+    The **Databases** view displays.
```
Comment (rmloveland) [inline cockroachcloud/backups-page.md:92]: Took me a sec to figure out what this is saying.  I guess in my mind I might phrase it as "The Databases view is displayed" - but perhaps that is a passive voice that we are avoiding?

(Aside: makes me think of "A wild **databases** view appears ..."

File: cockroachcloud/backups-page.md [hunk 3]
```diff
@@ -69,19 +71,95 @@ For each table in the database, the following details display:
...
+    The **Restore database** module displays with backup details.
+
+1. In the **Restore to** field, enter the name of the destination database.
+
+    {{site.data.alerts.callout_info}}
+    [Resolve any naming conflicts](#resolve-a-database-naming-conflict) by using `DROP` or `RENAME` on the existing database. If you enter a unique name in the **Restore to** field, a new database will be created.
+    {{site.data.alerts.end}}  
+
+1. Select any of the **Dependency options** to skip. You can:
+    - **Skip missing foreign keys**, which will remove [foreign key](../stable/foreign-key.html) constraints before restoring.
```
Comment (rmloveland) [inline cockroachcloud/backups-page.md:105]: Dumb question: does it remove FKs?  or just "skip missing ones" as the other text says?

Comment (rmloveland) [reply cockroachcloud/backups-page.md:105]: if it does remove FKs, does it remove some or all? we should say which, I think

File: cockroachcloud/backups-page.md [hunk 4]
```diff
@@ -69,19 +71,95 @@ For each table in the database, the following details display:
...
+    The **Tables** view displays.
+
+1. Click **Restore** for the table you want to restore.
+
+    The **Restore table** module displays with backup details.
+
+1. In the **Restore to** field, enter the name of the destination database.
+
+    {{site.data.alerts.callout_info}}
+    [Resolve any naming conflicts](#resolve-a-table-naming-conflict) by using `DROP` or `RENAME` on the existing table. If you enter a unique name in the **Restore to** field, a new table will be created.
```
Comment (rmloveland) [inline cockroachcloud/backups-page.md:136]: Same as above re: linking to SQL statement docs for these stmts

File: cockroachcloud/backups-page.md [hunk 5]
```diff
@@ -69,19 +71,95 @@ For each table in the database, the following details display:
...
+    {{site.data.alerts.end}}  
+
+1. Select any of the **Dependency options** to skip. You can:
+    - **Skip missing foreign keys**, which will remove [foreign key](../stable/foreign-key.html) constraints before restoring.
```
Comment (rmloveland) [inline cockroachcloud/backups-page.md:140]: Same as above re: FKs and exact behavior?

File: cockroachcloud/backups-page.md [hunk 6]
```diff
@@ -69,19 +71,95 @@ For each table in the database, the following details display:
...
+1. While [connected to your self-hosted CockroachDB cluster](../stable/connect-to-the-database.html), [back up](../stable/backup.html) your databases and/or tables to an [external location](../stable/backup.html#backup-file-urls):
 
     {% include copy-clipboard.html %}
     ~~~ sql
     > BACKUP DATABASE example_database TO 'gs://bucket_name/path_to_backup?AUTH=specified';
     ~~~
 
     {{site.data.alerts.callout_danger}}
-    If you are backing up the data to AWS or GCP, use the `specified` option for the `AUTH` parameter.
+    If you are backing up the data to AWS or GCP, use the `specified` option for the `AUTH` parameter, as CockroachCloud will need the `specified` credentials upon `RESTORE`.
```
Comment (rmloveland) [inline cockroachcloud/backups-page.md:162]: link to `RESTORE`?

File: cockroachcloud/backups-page.md [hunk 7]
```diff
@@ -122,9 +200,57 @@ Additionally, you can [back up and restore](../v20.1/backup-and-restore.html) yo
...
     {% include copy-clipboard.html %}
     ~~~ sql
     > RESTORE DATABASE example_database FROM 'gs://bucket_name/path_to_backup?AUTH=specified';
     ~~~
+
+## Troubleshooting
+
+### Resolve a database naming conflict
+
+The databases you want to restore cannot have the same name as an existing database in the target cluster. Before you restore a database, verify that the database name is not already in use. To do this, connect to your cluster with the [CockroachDB SQL client](connect-to-your-cluster.html#use-the-cockroachdb-sql-client) and run the following:
```
Comment (rmloveland) [inline cockroachcloud/backups-page.md:214]: might slightly edit "connect to your cluster" with "connect to the target cluster" for specificity

File: cockroachcloud/backups-page.md [hunk 8]
```diff
@@ -122,9 +200,57 @@ Additionally, you can [back up and restore](../v20.1/backup-and-restore.html) yo
...
+Or [change the existing database's name](../stable/rename-database.html):
+
+{% include copy-clipboard.html %}
+~~~ sql
+> ALTER DATABASE example_database RENAME TO archived_example_database;
+~~~
+
+### Resolve a table naming conflict
+
+The table you want to restore cannot have the same name as an existing table in the target database. Before you restore a table, verify that the table name is not already in use. To do this, connect to your cluster with the [CockroachDB SQL client](connect-to-your-cluster.html#use-the-cockroachdb-sql-client) and run the following:
```
Comment (rmloveland) [inline cockroachcloud/backups-page.md:237]: same as above: might slightly edit "connect to your cluster" with "connect to the target cluster" for specificity

**Review by rmloveland — APPROVED** [review]
LGTM with some suggestions (modulo the broken build - I think the build machine may not like the `stable` folder)

**Review by lnhsingh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/9292)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @j-low, @lnhsingh, @mwang1026, and @rmloveland)

___
*[cockroachcloud/backups-page.md, line 92 at r4](https://reviewable.io/reviews/cockroachdb/docs/9292#-MQJ8H275e2VCdnV5xdJ-r4-92:-MQs2Ym26kLLtTXk8qx_:bkdt8o6) ([raw file](https://github.com/cockroachdb/docs/blob/781e2edc8dd870f953dcdda1f79aa9c008beb2b5/cockroachcloud/backups-page.md#L92)):*
<details><summary><i>Previously, rmloveland (Rich Loveland) wrote…</i></summary><blockquote>

Took me a sec to figure out what this is saying.  I guess in my mind I might phrase it as "The Databases view is displayed" - but perhaps that is a passive voice that we are avoiding?

(Aside: makes me think of "A wild **databases** view appears ..."
</blockquote></details>

Yeah, was trying to avoid the passive voice. I could just remove this and put "In the Databases view, click Restore..." in the next step.
___
*[cockroachcloud/backups-page.md, line 101 at r4](https://reviewable.io/reviews/cockroachdb/docs/9292#-MQJ8H275e2VCdnV5xdJ-r4-101:-MQs2EJT1ST6uFySMSRO:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/781e2edc8dd870f953dcdda1f79aa9c008beb2b5/cockroachcloud/backups-page.md#L101)):*
<details><summary><i>Previously, rmloveland (Rich Loveland) wrote…</i></summary><blockquote>

links to `DROP DATABASE` and `RENAME DATABASE` ?
</blockquote></details>

Done.
___
*[cockroachcloud/backups-page.md, line 105 at r4](https://reviewable.io/reviews/cockroachdb/docs/9292#-MQJ8H275e2VCdnV5xdJ-r4-105:-MQtBdOn0l1yEso2qLzQ:byebp71) ([raw file](https://github.com/cockroachdb/docs/blob/781e2edc8dd870f953dcdda1f79aa9c008beb2b5/cockroachcloud/backups-page.md#L105)):*
<details><summary><i>Previously, rmloveland (Rich Loveland) wrote…</i></summary><blockquote>

if it does remove FKs, does it remove some or all? we should say which, I think
</blockquote></details>

It'll remove the missing FK constraints - added the info. @mwang1026 can you double check this to make I understand this as discussed in slack?
___
*[cockroachcloud/backups-page.md, line 136 at r4](https://reviewable.io/reviews/cockroachdb/docs/9292#-MQJ8H275e2VCdnV5xdJ-r4-136:-MQs2SuB2HU5GbixTq2R:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/781e2edc8dd870f953dcdda1f79aa9c008beb2b5/cockroachcloud/backups-page.md#L136)):*
<details><summary><i>Previously, rmloveland (Rich Loveland) wrote…</i></summary><blockquote>

Same as above re: linking to SQL statement docs for these stmts
</blockquote></details>

Done.
___
*[cockroachcloud/backups-page.md, line 162 at r4](https://reviewable.io/reviews/cockroachdb/docs/9292#-MQJ8H275e2VCdnV5xdJ-r4-162:-MQs5qyI56cpQ6gp755A:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/781e2edc8dd870f953dcdda1f79aa9c008beb2b5/cockroachcloud/backups-page.md#L162)):*
<details><summary><i>Previously, rmloveland (Rich Loveland) wrote…</i></summary><blockquote>

link to `RESTORE`?
</blockquote></details>

Done.
___
*[cockroachcloud/backups-page.md, line 214 at r4](https://reviewable.io/reviews/cockroachdb/docs/9292#-MQJ8H275e2VCdnV5xdJ-r4-214:-MQs7WiLDRPi4TMU_X4x:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/781e2edc8dd870f953dcdda1f79aa9c008beb2b5/cockroachcloud/backups-page.md#L214)):*
<details><summary><i>Previously, rmloveland (Rich Loveland) wrote…</i></summary><blockquote>

might slightly edit "connect to your cluster" with "connect to the target cluster" for specificity
</blockquote></details>

Done.
___
*[cockroachcloud/backups-page.md, line 237 at r4](https://reviewable.io/reviews/cockroachdb/docs/9292#-MQJ8H275e2VCdnV5xdJ-r4-237:-MQs7Zqg8YC9bg_zaA22:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/781e2edc8dd870f953dcdda1f79aa9c008beb2b5/cockroachcloud/backups-page.md#L237)):*
<details><summary><i>Previously, rmloveland (Rich Loveland) wrote…</i></summary><blockquote>

same as above: might slightly edit "connect to your cluster" with "connect to the target cluster" for specificity
</blockquote></details>

Done.

**Pull request merged**